pkgs/applications/audio/bambootracker/default.nix

```diff
@@ -0,0 +1,55 @@
+{ mkDerivation, stdenv, fetchFromGitHub
+, qmake, qtbase, qtmultimedia, qttools
+, alsaSupport ? stdenv.hostPlatform.isLinux, alsaLib
+, pulseSupport ? stdenv.hostPlatform.isLinux, libpulseaudio
+, jackSupport ? stdenv.hostPlatform.isUnix, libjack2
+}:
+
+let
+
+  opt = stdenv.lib.optional;
+
+in mkDerivation rec {
+  pname = "bambootracker";
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "rerrahkr";
+    repo = "BambooTracker";
+    rev = "v${version}";
+    sha256 = "06ij0k5sfns1zakl6qbxr5j91dj1kfj3s5jc3mpqzxc3zpc3avkv";
+  };
+
+  sourceRoot = "source/BambooTracker";
+
+  nativeBuildInputs = [ qmake qttools ];
+
+  buildInputs = [ qtbase qtmultimedia ]
+  ++ opt alsaSupport alsaLib
+  ++ opt pulseSupport libpulseaudio
+  ++ opt jackSupport libjack2;
+
+  qmakeFlags = []
+  ++ opt alsaSupport  "DEFINES+=__LINUX_ALSA__"
+  ++ opt pulseSupport "DEFINES+=__LINUX_PULSE__"
+  ++ opt jackSupport  "DEFINES+=__UNIX_JACK__";
+
+  postInstall = ''
+    install -d $out/share/{man/man1,applications,BambooTracker,icons/hicolor/{16x16,256x256}/apps}
+    install -t $out/share/man/man1/      ../BambooTracker*.1
+    install -t $out/share/applications/  ../BambooTracker.desktop
+    cp -at     $out/share/BambooTracker/ ../demos
+
+    for size in 16x16 256x256; do
+      install res/icon/icon_$size.png $out/share/icons/hicolor/$size/apps/BambooTracker.png
```

Comment:
`install -m644` on non-executables.

You may also want to use `-D` and skip the `-d` step, so those lines don't need to be kept in sync

Reply:
Gonna change it to `install -Dm644` for the icons & desktop file.

I saw `installManPage` being used in a recent PR, should I use that instead for the manpages?

For the demos, I don't know how to `install` a whole directory with subdirectories. Should I just `chmod 644 -R $out/share/BambooTracker/demos` after `cp`ing them over?

Reply:
> installManPage

I never had to install a manpage yet, but I guess if something like this exists, using it is a goo idea.

> a whole directory
Just `cp -r` is fine afaik. No need to chmod unless the dev set them wrong in the first place I think.

Reply:
Fixed.